test.py

```diff
@@ -17,10 +17,54 @@
 from omcommon import friendlyStr, multiple_replace
 from natsort import natsorted
 from shared import readConfig, getReferenceFileName, simulationAcceptsFlag, isFMPy
+from platform import processor
 import shared
 
 import signal
 
+parser = argparse.ArgumentParser(description='OpenModelica library testing tool')
+parser.add_argument('configs', nargs='*')
+parser.add_argument('--branch', default='master')
+parser.add_argument('--fmi', default=False)
+parser.add_argument('--output', default='')
+parser.add_argument('--docker', default='')
+parser.add_argument('--libraries', help="Directory omc will search in to load system libraries/libraries to test.", default=os.path.normpath(os.path.expanduser('~/.openmodelica/libraries/')))
+parser.add_argument('--extraflags', default='')
+parser.add_argument('--extrasimflags', default='')
+parser.add_argument('--ompython_omhome', default='')
+parser.add_argument('--noclean', action="store_true", default=False)
+parser.add_argument('--fmisimulator', default='')
+parser.add_argument('--ulimitvmem', help="Virtual memory limit (in kB) (linux only)", type=int, default=8*1024*1024)
+parser.add_argument('--default', action='append', help="Add a default value for some configuration key, such as --default=ulimitExe=60. The equals sign is mandatory.", default=[])
+parser.add_argument('-j', '--jobs', default=0)
+parser.add_argument('-v', '--verbose', action="store_true", help="Verbose mode.", default=False)
+parser.add_argument('--execAllTests', action="store_true", help="Force all tests to be executed", default=False)
+parser.add_argument('--win', action="store_true", help="Windows mode", default=False)
+parser.add_argument('--noSync', action="store_true", help="Move files using python instead of rsync", default=False)
+parser.add_argument('--timeout', default=0, help="=[value] timeout in seconds for each test, it overrides the timeout calculated by the script")
+
+args = parser.parse_args()
```

Comment:
While you are touching this code, maybe it's preferable to use the opportunity to clean up the structure of this script a bit, and put the argument parsing into a separate function that returns the `args` structure? Thus you reduce the amount of code that just sits at the top level, and proceed towards decoupling the separate bits a little.

Reply:
shure, I'll implement both the suggestions